### PR TITLE
Fix kvp test case failures

### PR DESCRIPTION
**Pull request by squirrelsc**
See details in commits.

### CODE REVIEW

File: microsoft/testsuites/core/kvp.py [hunk 1]
```diff
@@ -31,7 +31,8 @@
 )
 class Kvp(TestSuite):
     # lrwx------ 1 root root 64 Oct 18 19:05 9 -> /dev/vmbus/hv_kvp
-    _hv_kvp_pattern = re.compile(r".* /dev/vmbus/(hv_kvp)$", re.M)
+    # lrwx------. 1 root root 64 Dec 18 07:38 4 -> /dev/vmbus/hv_kvp
+    _hv_kvp_pattern = re.compile(r".* /dev/vmbus/(hv_kvp)\r?$", re.M)
```
Comment (LiliDeng) [inline microsoft/testsuites/core/kvp.py:34]: is a new line character in the second sample?

File: lisa/sut_orchestrator/azure/tools.py [hunk 2]
```diff
@@ -225,9 +225,9 @@ def get_pool_records(self, pool_id: int, force_run: bool = False) -> Dict[str, s
         result = self.run(
             str(pool_id),
             force_run=force_run,
-            expected_exit_code=0,
-            expected_exit_code_failure_message="failed to get pool",
         )
+        # some distro return 4, for example, Ubuntu Server 1804
+        assert_that(result.exit_code).described_as("failed to get pool").is_in(0, 4)
```
Comment (LiliDeng) [inline lisa/sut_orchestrator/azure/tools.py:230]: we can add one more comment after figuring out what the error code 4 stands for.

**Comment by LiliDeng** [comment]
Also, as we know some distros don't have kvp by default in azure platform, if it is expected, then we need mark case as skipped in step 1?

File: lisa/sut_orchestrator/azure/tools.py (hunk 2 above)
Comment (squirrelsc) [reply lisa/sut_orchestrator/azure/tools.py:230]: I don't know what code 4 is here, and it succeeds. So I just record the distro name here.

File: microsoft/testsuites/core/kvp.py (hunk 1 above)
Comment (squirrelsc) [reply microsoft/testsuites/core/kvp.py:34]: No, I thought its format is different, but actually the reason is the \r at the end. I will remove the second sample.

**Comment by squirrelsc** [comment]
> Also, as we know some distros don't have kvp by default in azure platform, if it is expected, then we need mark case as skipped in step 1?

will mark them in next round failure analysis.

**Review by LiliDeng — APPROVED** [review]

**Pull request merged**